### ISSUE TITLE
[FEATURE] - c

### ISSUE THREAD

**Issue by hanekoo** [tqwewe/kameo#75]
## Feature Description
A clear and concise description of the feature or enhancement you'd like to see.

## Motivation
Explain why this feature is important or useful. What problem does it solve or what use case does it improve?

## Proposed Solution
If you have an idea of how to implement this feature, please provide a high-level overview of the solution.

## Alternatives Considered
Have you considered any alternative solutions? If so, please describe them.

## Additional Context
Add any other context, screenshots, or examples that would help illustrate the feature.

**Comment by tqwewe** [comment]
Thanks for your feedback, very valuable!

1. `on_link_start` is a very nice idea I haven't thought of. Did you have a particular use case for this in your experience with kameo?

2. I'd still like to explore the possibility of using a broadcast mailbox, however I couldn't find a nice API design I was happy with. But its something I'm open to for suggestions and ideas

3. Improving the usage of PubSub would be great, and having it be a single message type is a little limiting. Thanks for pointing out actix broker, I haven't seen that one before

All great suggestions, and I'll try exploring these when I get more time, thank you :)

**Comment by marcaddeo** [comment]
Something tangentially related I've been thinking about; would it be possible for actors to know about their parent, child, sibling actor refs when actors are supervised/linked? Is that even a good idea? 

When I first started using Kameo, that's how I expected the parent/child supervision to behave (I'm not sure whether or not this is how Erlang/OTP function) but ended up realizing I need to pass the refs manually between the actors that need to communicate with each other.

**Comment by tqwewe** [comment]
Would an `on_linked` or `on_link_start` hook added to the `Actor` trait solve this @marcaddeo?

**Comment by hanekoo** [comment]
> Something tangentially related I've been thinking about; would it be possible for actors to know about their parent, child, sibling actor refs when actors are supervised/linked? Is that even a good idea?我一直在思考一些与此无关的事情；当演员受到监督/链接时，演员是否有可能知道他们的父母、孩子、兄弟演员参考资料？这是个好主意吗？
> 
> When I first started using Kameo, that's how I expected the parent/child supervision to behave (I'm not sure whether or not this is how Erlang/OTP function) but ended up realizing I need to pass the refs manually between the actors that need to communicate with each other.当我第一次开始使用 Kameo 时，这就是我期望父/子监督的行为方式（我不确定这是否是 Erlang/OTP 的功能），但最终意识到我需要在参与者之间手动传递引用需要互相沟通。

Yes, I've encountered this too.

- Sometimes it's uncertain whether an actor reference is needed within a struct.
- It can also be intrusive to the struct.

**Comment by hanekoo** [comment]
> 1. `on_link_start` is a very nice idea I haven't thought of. Did you have a particular use case for this in your experience with kameo?`on_link_start`是一个非常好的主意，我没有想到。根据您使用 kameo 的经验，您是否有过特殊的用例？

This mainly depends on the application’s design requirements, especially in applications where actors have strong dependencies on each other.

For example, separating an actor that initializes a `TcpStream` (`init_actor`) from the one that processes the `TcpStream` (`process_actor`), but with dependencies between them. `process_actor` depends on whether `init_actor` starts successfully (though this example may be a bit of a stretch).

**Comment by marcaddeo** [comment]
> Would an `on_linked` or `on_link_start` hook added to the `Actor` trait solve this @marcaddeo?

Yeah, if that hook contained the actor ref and some sort of "`LinkKind::Child`" marker to know what kind of link it is. I think that would probably work great.

**Comment by tqwewe** [comment]
@marcaddeo I've started exploring the addition of an `on_linked` hook to handle scenarios where an actor is linked with another. The challenge, however, is that the `ActorRef` for the other actor can't be strongly typed within the trait definition, as it depends on the specific actor being linked. Unfortunately, we also can't use `ActorRef<Box<dyn Actor>>` due to type constraints.

As a result, the best we can provide in the `on_linked` hook is the other actor's `ActorID`. That said, this limitation makes me question how practical or useful the hook would be in its current form.
